src/browser/file-recovery-service.js

```diff
@@ -0,0 +1,65 @@
+'use babel'
+
+import {ipcMain} from 'electron'
+import crypto from 'crypto'
+import Path from 'path'
+import fs from 'fs-plus'
+
+export default class FileRecoveryService {
+  constructor (recoveryDirectory) {
+    this.recoveryDirectory = recoveryDirectory
+    this.recoveryPathsByWindowAndFilePath = new WeakMap()
+    this.crashListeners = new WeakSet()
+  }
+
+  start () {
+    ipcMain.on('will-save-path', this.willSavePath.bind(this))
+    ipcMain.on('did-save-path', this.didSavePath.bind(this))
```

Comment:
Should we keep a `CompositeDisposable` for removing these listeners? Might not be a big deal since everything will be torn down when the main process is killed, but might be good to clean up for testing purposes?